### ISSUE TITLE
New check: LambdaParameterName

### ISSUE THREAD

**Issue by tsjensen** [checkstyle/checkstyle#3009]
I would like to propose a new check _LambdaParameterName_ for the naming convention on lambda arguments. Just like [ParameterName](http://checkstyle.sourceforge.net/config_naming.html#ParameterName), it would be a subclass of `AbstractNameCheck` and have a `format` property.

Having this check would allow for a more precise definition of the naming conventions. Currently, lambda arguments must follow the same naming convention as method arguments. But some people prefer to make a distinction between the two, so that one knows which parameter came from the enclosing method and which from the lambda expression:

``` java
public void method(Person pPerson) {
  printAdultsWithSameGender(roster,
    (Person p) -> p.getGender() == pPerson.getGender() && p.getAge() >= 18
  );
}
```

For compatibility reasons, we should probably add a new property `ignoreLambdas` to the existing _ParameterName_ check which is by default set to `false`. Then people can set it to `true` when they introduce the new check to their rule sets.
##

**Comment by romani** [comment]
I can not aprove that for now, as we are still in moratorium period for new Checks. I accepting new Checks only in sevntu project.

**Comment by tsjensen** [comment]
Ok, never mind then.
(Interesting that does not seem to be the case for some other checks.)

**Comment by romani** [comment]
Yes, you can fix 5 functional issues and introduce any new Check. I thought that already know that https://groups.google.com/forum/m/#!topic/checkstyle-devel/JbhfiiSqKsQ . You can reopen this issue to us remember that was requested already. In sevntu project you can provide implementation, that will be moved to main repo as soon as we get out of moratorium.

**Comment by tsjensen** [comment]
I did not offer to help, just proposed an idea. Don't worry! :-) I'm sure you'll find mister right to do the implementation one day. I'll reopen the issue as a reminder like you suggested.

**Comment by rnveach** [comment]
I am closing this issue as it was completed in https://github.com/checkstyle/checkstyle/issues/5893
